Sources/Core/Views/GenericButton/GenericButton.swift

```diff
@@ -0,0 +1,176 @@
+//
+//  GenericButton.swift
+//  PovioKit
+//
+//  Created by Arlind Dushi on 3/17/22.
+//  Copyright © 2022 Povio Inc. All rights reserved.
+//
+
+import SwiftUI
+
+@available(iOS 13, *)
+public class GenericButtonProperties: ObservableObject {
+  @Published public var title: String = ""
+  @Published public var font: Font = .system(size: 18)
+  @Published public var textColor: Color = .white
+  @Published public var cornerRadius: GenericButton.CornerRadiusType = .custom(0)
+  @Published public var backgroundType = GenericButton.Background.plain(.blue)
+  @Published public var borderColor: Color = .clear
+  @Published public var borderWidth: CGFloat = 0
+  @Published public var extraImage: GenericButton.ExtraImage?
+}
+
+@available(iOS 13, *)
+public struct GenericButton: View {
+  @ObservedObject public var properties = GenericButtonProperties()
```

Comment:
I would suggest here to change viewModel to private property and to implement Builder pattern for applying properties:
`@ObservedObject private var viewModel = GenericButtonViewModel()`

For every property we will have separate Builder method like this:
```
//MARK:- Builder Pattern Methods
  func title(_ title: String) -> GenericButton {
    self.viewModel.title = title
    return self
  }
  
  func font(_ font: Font) -> GenericButton {
    self.viewModel.font = font
    return self
  }
  // add rest of the properties
```

And later when we add them in code it will be more in SwiftUI style:

```
GenericButton()
      .backgroundType(.linearGradient(
        LinearGradient(gradient: Gradient(colors: [.blue, .green]), startPoint: .top, endPoint: .bottom)))
      .cornerRadius(.rounded)
      .borderWidth(3)
      .borderColor(.red)
      .extraImage(.right(Image(systemName: "arrow.right")))
      .title("Test")
      .frame(width: 200, height: 70)
```
What do you think @ArlindDushi ?

Reply:
I like this proposal, it makes it more SwiftUI style. I thought about this but the only small gripe I have with it is that if we use it in `UIKit` after we declared the button in global scope  ( b/c we need to expose it for whatever reason which happens very often), then to modify properties we have to use it like this:  

```
_ = actionButton
      .title("Sign up")
```

otherwise it'd create a `Result of call to 'title' is unused` 


I guess this is where we have to decide if we want to keep it more `UIKit` style for now or go with a more `SwiftUI` approach.  I have no strong feelings towards this. 

What do you think @borut-t , @poviolabs/ios-developers ?

Reply:
Let’s go with this solution since we are still more likely to use UIKit...